crates/revm/revm-inspectors/src/stack/maybe_owned.rs

```diff
@@ -85,86 +90,92 @@ where
 
     fn log(
         &mut self,
-        evm_data: &mut EVMData<'_, DB>,
+        context: &mut EvmContext<'_, DB>,
         address: &Address,
         topics: &[B256],
         data: &Bytes,
     ) {
         match self {
             MaybeOwnedInspector::Owned(insp) => {
-                return insp.borrow_mut().log(evm_data, address, topics, data)
+                return insp.borrow_mut().log(context, address, topics, data)
             }
             MaybeOwnedInspector::Stacked(_) => {}
         }
     }
 
-    fn step_end(&mut self, interp: &mut Interpreter<'_>, data: &mut EVMData<'_, DB>) {
+    fn step_end(&mut self, interp: &mut Interpreter, context: &mut EvmContext<'_, DB>) {
         match self {
-            MaybeOwnedInspector::Owned(insp) => insp.borrow_mut().step_end(interp, data),
+            MaybeOwnedInspector::Owned(insp) => insp.borrow_mut().step_end(interp, context),
             MaybeOwnedInspector::Stacked(_) => {}
         }
     }
 
     fn call(
         &mut self,
-        data: &mut EVMData<'_, DB>,
+        context: &mut EvmContext<'_, DB>,
         inputs: &mut CallInputs,
-    ) -> (InstructionResult, Gas, Bytes) {
+    ) -> Option<(InterpreterResult, Range<usize>)> {
         match self {
-            MaybeOwnedInspector::Owned(insp) => return insp.borrow_mut().call(data, inputs),
+            MaybeOwnedInspector::Owned(insp) => return insp.borrow_mut().call(context, inputs),
             MaybeOwnedInspector::Stacked(_) => {}
         }
 
-        (InstructionResult::Continue, Gas::new(0), Bytes::new())
+        Some((
+            InterpreterResult {
+                result: InstructionResult::Continue,
+                gas: Gas::new(0),
+                output: Bytes::new(),
+            },
+            0..0,
```

Comment:
@rakita What does this new range in the result here signify? Tests are passing, though I don't think this is correct. Occurs in several places.

Reply:
If you want to shortcircuit the call you can return the result. So this should be `None`